sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/Makefile.include

```diff
@@ -47,28 +49,52 @@ RISCV_NM       ?= ${OT_TOOLS}/riscv32-unknown-elf-nm
 RISCV_READELF  ?= ${OT_TOOLS}/riscv32-unknown-elf-readelf
 RISCV_GCC_OPTS ?= -static -mcmodel=medany -fvisibility=hidden -nostdlib -nostartfiles -g
 
+BUILD_ROOT = $(work_dir_isa)/meson-build
 
 # The compile target recipe re-uses the boot rom library.
 # This will be changed in the future when the compliance tests directly build
 # their own libraries
 # After the libraries are built, the necessary collateral (vmem for verilator, bin
 # for fpga) are created
 COMPILE_TARGET=\
-        make -C $$(OT_SW) SW_DIR=boot_rom $(MAKEARG) RV_TOOLS=$(OT_TOOLS) clean all; \
+  export BUILD_ROOT=$(BUILD_ROOT); \
+	mkdir -p $(BUILD_ROOT); \
+	cd $(OT_ROOT); \
+	./meson_init.sh; \
+	cd -; \
+	ninja -C "$(BUILD_ROOT)/build-out/sw/$(NINJA_TARGET)" all; \
+	\
 	$$(RISCV_GCC) $(2) $$(RISCV_GCC_OPTS) \
+		-I$(OT_ROOT) \
 		-I$(ROOTDIR)/riscv-test-env/ \
 		-I$(ROOTDIR)/riscv-test-env/p/ \
-                -I$(OT_SW)/lib \
-                -I$(OT_ROOT)/build/sw/device/boot_rom/lib \
 		-I$(TARGETDIR)/$(RISCV_TARGET)/ \
-		-I$(TARGETDIR)/$(RISCV_TARGET)/ \
-		$(DEFINES) -T$(LDSCRIPT) $$< \
-                $(OPENTITAN)/wrap.c \
-                -L$(OT_ROOT)/build/sw/device/boot_rom/lib \
-                -lot \
+		$(DEFINES) \
+		-T$(LDSCRIPT) \
+		$$< \
+		$(OPENTITAN)/wrap.c \
+		# The line below does some technically-not-allowed peeking into build-out.
+		# This is a temporary workaround until we Meson-ify RISC-V compliance.
+		-L$(BUILD_ROOT)/build-out/sw/$(NINJA_TARGET)/sw/device/lib \
+		-luart_ot \
```

Comment:
I don't see an include line for where wrap.c will find "uart.h".  Is that handled by the library? I thought It needed to be specifically called out still.

Reply:
Include resolution in C is awful. Each -I argument adds a search path, relative to which #includes are resolved. It can find uart.h because:
- I made that include absolute.
- I included -I$(OT_ROOT).
- The `-L` and `-l` then cause libuart_ot.a to get linked in correctly.

If this sounds like an insane way to do things, that's because it is.

Reply:
ah sorry i get the -L / -l part.
It's more than in wrap.c i think...the reference to uart.h wasn't absolute (at least i didn't have it that way)

Reply:
ah i see you changed it. looks good.